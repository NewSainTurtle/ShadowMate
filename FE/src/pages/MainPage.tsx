import React, { useState } from "react";
import styles from "@styles/Main.module.scss";

const MainPage = () => {
  const [number, setNumber] = useState<number>(0);

  return (
    <>
      <div className={styles.MyComponent}>
        <h1 className={styles.header01}>Test Main</h1>
        <h2>Hello. Jest!!</h2>
        <h4>number: {number}</h4>
        <button onClick={() => setNumber(number + 3)}>증가</button>
        <button onClick={() => setNumber(number - 2)}>감소</button>
      </div>
    </>
  );
};

export default MainPage;
