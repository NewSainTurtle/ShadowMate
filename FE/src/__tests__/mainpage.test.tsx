import React from "react";
import MainPage from "@pages/MainPage";
import { describe, expect } from "@jest/globals";
import { fireEvent, render } from "@testing-library/react";

describe("MainPage test", () => {
  it("matches snapshopt", () => {
    const { container } = render(<MainPage />);
    expect(container).toMatchSnapshot(); // snapshot match
  });

  it("screen test", () => {
    const { container } = render(<MainPage />);
    const h2 = container.querySelector("h2"); // h2태그 DOM 가져오기
    expect(h2?.innerHTML).toBe("Hello. Jest!!"); // h2의 innerHTML이 Hello, Jest!! 인지 테스트
  });

  it("button test", () => {
    const utils = render(<MainPage />);

    const str = utils.getByText("number: 0"); // 텍스트가 number: 0인 DOM 가져오기
    const increaseButton = utils.getByText("증가"); // 텍스트가 증가인 DOM 가져오기
    const decreaseButton = utils.getByText("감소"); // 텍스트가 감소인 DOM 가져오기

    fireEvent.click(increaseButton); // 증가버튼 클릭이벤트 실행
    fireEvent.click(increaseButton); // 증가버튼 클릭이벤트 실행
    expect(str).toHaveTextContent("number: 6"); // str의 텍스트가 number: 6인지 확인

    fireEvent.click(decreaseButton); // 감소버튼 클릭이벤트 실행
    fireEvent.click(decreaseButton); // 감소버튼 클릭이벤트 실행
    expect(str).toHaveTextContent("number: 2"); // str의 텍스트가 number: 2인지 확인
  });
});
