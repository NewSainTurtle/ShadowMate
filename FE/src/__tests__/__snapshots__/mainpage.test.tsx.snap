// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage test matches snapshopt 1`] = `
<div>
  <div
    class="MyComponent"
  >
    <h1
      class="header01"
    >
      Test Main
    </h1>
    <h2>
      Hello. Jest!!
    </h2>
    <h4>
      number: 
      0
    </h4>
    <button>
      증가
    </button>
    <button>
      감소
    </button>
  </div>
</div>
`;
